deploy/local/vars_lnx.sh

```diff
@@ -15,77 +15,153 @@
 # limitations under the License.
 #===============================================================================
 
-# ############################################################################
-# Get absolute path to script, when sourced from bash, zsh and ksh shells.
-# Uses `readlink` to remove links and `pwd -P` to turn into an absolute path.
-# Derived from similar function used by VTune and Advisor.
-# Converted into a POSIX-compliant function.
+# shellcheck shell=sh
+# Get absolute pathname to script, when sourced from bash, zsh and ksh shells.
+# see https://stackoverflow.com/a/29835459/2914328 for detailed "how it works"
+#
+# Attribution of this rreadlink() function goes to Michael Klement, aka
+# "mklement0" on StackOverflow. Based on "edited Sep 16 '19" version of SO post.
+# Based on https://github.com/mklement0/rreadlink/blob/master/bin/rreadlink#L125
+# Above licensed under MIT license > https://spdx.org/licenses/MIT#licenseText
+#     User profile: https://stackoverflow.com/users/45375/mklement0
+#         LinkedIn: https://www.linkedin.com/in/mklement0/
+#          SO post: https://stackoverflow.com/a/29835459/2914328
+# SO License terms: https://stackoverflow.com/help/licensing
+#     CC BY-SA 4.0: https://creativecommons.org/licenses/by-sa/4.0/
+#
+# Adapted and modified by Paul A. Fischer, Intel Corporation
+#
+# This POSIX-compliant shell function implements an equivalent to the GNU
+# `readlink -e` command and is a reasonably robust solution that only fails
+# in two rare edge cases:
+#   * paths with embedded newlines (very rare)
+#   * filenames containing literal string -> (also rare)
 
 # Usage:
-#   script_dir=$(get_script_path "$script_rel_path")
+#   script_path=$(rreadlink "$vars_script_rel_path")
+#   script_dir_path=$(dirname -- "$(rreadlink "$vars_script_rel_path")")
 #
 # Inputs:
 #   script/relative/pathname/scriptname
 #
 # Outputs:
-#   /script/absolute/pathname
+#   /script/absolute/pathname/scriptname
+
 # executing function in a *subshell* to localize vars and effects on `cd`
-get_script_path() (
-  script="$1"
-  while [ -L "$script" ] ; do
-    # combining next two lines fails in zsh shell
-    script_dir=$(command dirname -- "$script")
-    script_dir=$(cd "$script_dir" && command pwd -P)
-    script="$(readlink "$script")"
-    case $script in
-      (/*) ;;
-       (*) script="$script_dir/$script" ;;
-    esac
+rreadlink() (
```

Comment:
I see a lot of the same code in two scripts for mac and lnx. Is it possible to use the common part? Because any change will be very expensive

Reply:
this is question to merging separate scripts and updating makefile to use - i don't want to handle this as we also have feebsd env file